facts/get_facts.sh

```diff
@@ -9,6 +9,20 @@ apt_install() {
                 install --yes --force-yes "$@"
 }
 
+write_facts() {
+  output_file="$1"
+  mkdir -p $(dirname ${output_file})
+  facter --show-legacy -p -j | tee ${output_file}
+}
+
+write_facts_with_major() {
+  write_facts "/vagrant/$(facter --version | cut -d. -f1,2)/$(facter operatingsystem | tr '[:upper:]' '[:lower:]')-$(facter operatingsystemmajrelease)-$(facter hardwaremodel).facts"
```

Comment:
Suggest adding a comment to explain what is happening here and why.

Reply:
I left it out of this PR for now.